src/uwtools/drivers/jedi.py

```diff
@@ -0,0 +1,134 @@
+"""
+A driver for the jedi component.
+"""
+
+import logging
+from datetime import datetime
+from pathlib import Path
+
+from iotaa import asset, dryrun, run, task, tasks
+
+from uwtools.config.formats.yaml import YAMLConfig
+from uwtools.drivers.driver import Driver
+from uwtools.strings import STR
+from uwtools.utils.tasks import file, filecopy, symlink
+
+
+class JEDI(Driver):
+    """
+    A driver for the JEDI component.
+    """
+
+    def __init__(self, config: Path, cycle: datetime, dry_run: bool = False, batch: bool = False):
+        """
+        The driver.
+
+        :param config: Path to config file.
+        :param cycle: The forecast cycle.
+        :param dry_run: Run in dry-run mode?
+        :param batch: Run component via the batch system?
+        """
+        super().__init__(config=config, dry_run=dry_run, batch=batch, cycle=cycle)
+        if self._dry_run:
+            dryrun()
+        self._cycle = cycle
+
+    # Workflow tasks
+
+    @tasks
+    def files_copied(self):
+        """
+        Files copied for run.
+        """
+        yield self._taskname("files copied")
+        yield [
+            filecopy(src=Path(src), dst=self._rundir / dst)
+            for dst, src in self._driver_config.get("files_to_copy", {}).items()
+        ]
+
+    @tasks
+    def files_linked(self):
+        """
+        Files linked for run.
+        """
+        yield self._taskname("files linked")
+        yield [
+            symlink(target=Path(target), linkname=self._rundir / linkname)
+            for linkname, target in self._driver_config.get("files_to_link", {}).items()
+        ]
+
+    @tasks
+    def provisioned_run_directory(self):
+        """
+        Run directory provisioned with all required content.
+        """
+        yield self._taskname("provisioned run directory")
+        yield [
+            self.files_copied(),
+            self.files_linked(),
+            self.configuration_file(),
+            self.runscript(),
+        ]
+
+    @task
+    def runscript(self):
+        """
+        The runscript.
+        """
+        path = self._runscript_path
+        yield self._taskname(path.name)
+        yield asset(path, path.is_file)
+        yield None
+        self._write_runscript(path=path, envvars={})
+
+    @task
+    def validate_only(self):
+        """
+        Validate config.
```

Comment:
I think users might benefit from seeing a slightly more thorough one-line description of this task.

Also, at the moment, I think this task would

1. Only be executed when explicitly requested by the user, and
2. Only validate the `base_file`, before any `update_values` are applied to it.

I'm not sure, but is our idea that `base_file` is likely to be something supplied by JEDI, or perhaps a version of that file modified by an app? If that's the case, it seems likely that this file will usually be valid, and that it might be useful to also validate the file created by the `configuration_file` task, after applying the `update_values` supplied by the user. If I'm not on the wrong track here, maybe it'd be useful to make this task depend on `configuration_file()` instead of `file(base_file)` (ok, it actually says `file(path)` here now, but I'll add a suggestion to change `path` to `base_file` to make it clearer what that file is) and add it as a requirement of `provisioned_run_directory` in place of `configuration_file` (since it transitively requires `configuration_file`)... and/or make it a requirement of `run` so that the config file is always validated before a run is performed.

Reply:
A couple questions on this suggestion. 

There seems to be a couple parts to this change:
1. Add a dependency for `validate_only` to depend on `configuration_file`
2. Change `path` to `base_file` where appropriate
3. Add the requirement of `base_file` to `provisioned_run_directory`


Questions:
1. In regards to the first change, I was trying to find some documentation on iotaa to create a dependency like this. Does this require a separate `@tasks` method that handles this dependency? Can this dependency be addressed by `validate_only` calling `configuration_file` and waiting for the contents back in a `yield` statement? Here I was thinking we are calling `self.configuration_file()` instead of `self._driver_config["configuration_file"]["base_file"]` and storing this value in the `base_file` variable. I was thinking about accessing the values within the returned config from the method, but it seems like it is not a scriptable and without any value it gives an error of `no file or directory to "/path/to/jedi.yaml"` which is in our tests. 
Trying to access this correctly could be the issue i.e., `base_file = Path(self.configuration_file["configuration_file"]["base_file"])` vs. `base_file = Path(self.configuration_file()["configuration_file"]["base_file"])`

```
    @task
    def validate_only(self):
        """
        Validate the JEDI driver config
        """
        taskname = self._taskname("validate_only")
        yield taskname
        a = asset(None, lambda: False)
        yield a
        base_file = Path(self.configuration_file["configuration_file"]["base_file"])
        executable = Path(self._driver_config["execution"]["executable"])
        yield [file(executable), file(base_file)]
        env = " && ".join(self._driver_config["execution"]["envcmds"])
        cmd = "{env} && time {x} --validate-only {p} 2>&1".format(env=env, x=executable, p=path) 
        result = run(taskname, cmd)
        if result.success:
            logging.info("%s: Config is valid", taskname)
            a.ready = lambda: True
```

3. Add the requirement of `base_file` to `provisioned_run_directory`. To me this only makes sense when `base_file` is a `@task` itself. Unless we are trying to use that dependency with `configuration_file` again to do something like `self.configuration_file()["base_file"]